torch/_subclasses/fake_tensor.py

```diff
@@ -2589,3 +2599,66 @@ def dump_cache_stats() -> None:
         width = max(len(k) for k in bypasses)
         for k, v in sorted(bypasses.items(), key=lambda i: -i[1]):
             log.info("    %-*s %s", width + 1, f"{k}:", v)
+
+
+def inferred_fake_kernel_from_real_out(
+    mode: FakeTensorMode, op: torch._ops.OpOverload, real_out: Any
+) -> Any:
+    assert mode.shape_env is not None
+
+    # Only support operators that have all Tensor outputs
+    real_flat_out, spec = pytree.tree_flatten(real_out)
+    if not all(isinstance(t, torch.Tensor) for t in real_flat_out):
+        raise RuntimeError(
+            f"propagate_real_tensors: we don't support operators that return "
+            f"non-Tensors. Got {op._name}"
+        )
```

Comment:
could we just return the primitive values in this case? Besides tensors & those, is it just ScriptObjects that we'd have to deal with?

Reply:
If an operator returns a non-symbolic float or an int, we'd end up baking it into the graph. So the graph isn't useful for anything beyond the current sample inputs.

Also, most operators do not return non-symbolic floats or ints. If the data in the future ends up showing we need to relax this, we can. But I intended this PR to be an initial conservative version that we can expand on.